pkg/model/instance.go

```diff
@@ -35,18 +35,15 @@ type Linked struct {
 
 func (i Instance) FindParameter(name string) (InstanceParameter, error) {
 	for _, parameter := range i.Parameters {
-		if parameter.StackParameterID == name {
+		if parameter.StackParameterName == name {
 			return parameter, nil
 		}
 	}
 	return InstanceParameter{}, fmt.Errorf("parameter not found: %s", name)
 }
 
 type InstanceParameter struct {
-	InstanceID uint `json:"-" gorm:"primaryKey"`
-	// TODO: Rename StackParameterID to Name
-	StackParameterID string         `json:"name" gorm:"primaryKey"`
-	StackParameter   StackParameter `json:"-" gorm:"foreignKey:StackParameterID,StackName; references:Name,StackName; constraint:OnUpdate:CASCADE,OnDelete:CASCADE"`
-	StackName        string         `json:"-"`
-	Value            string         `json:"value"`
+	InstanceID         uint   `json:"-" gorm:"primaryKey"`
+	StackParameterName string `json:"name" gorm:"primaryKey"`
```

Comment:
* Since we are within the struct `InstanceParameter` we know that the field is one of a parameter.
* Since the `InstanceParameter` goes through validation we also know that it needs to match the stack the instance is associated with.

I would therefore like to keep it short and call this field `Name`. This would then also match the name in `StackParameter`.

Reply:
So would I! But previously I haven't been able to make it work due to a sql error... ambiguous column name or something, I don't recall the exact error